internal/docs/exported_fields.go

```diff
@@ -127,11 +127,20 @@ func visitFields(namePrefix string, f fields.FieldDefinition, records []fieldsTa
 
 	if len(f.Fields) == 0 && f.Type != "group" {
 		if f.External != "" {
-			var err error
-			f, err = fdm.ImportField(f.External, name)
+			imported, err := fdm.ImportField(f.External, name)
 			if err != nil {
 				return nil, errors.Wrap(err, "can't import field")
 			}
+			f.Type = imported.Type
```

Comment:
nit: WDYT about introducing a new method to extract these if-conditions? For example: `f.Update(imported)`

Reply:
Good idea :+1:

Reply:
Done in https://github.com/elastic/elastic-package/pull/575/commits/095d0ca37007747a338b502c0e94ab0b36a283bd.

Though doing it as a method of the `FieldDefinition` implies to do it for all its attributes, including the sub-fields. I have implemented this part by updating sub-fields with the same name, and appending the rest.

An alternative could have been to extract these if-conditions in a method here, not as a method of the struct, and then I could only update the fields relevant here.

Let me know what you think.